api/src/main/java/org/openmrs/api/impl/ConceptServiceImpl.java

```diff
@@ -1977,4 +1977,46 @@ public boolean hasAnyConceptAttribute(ConceptAttributeType conceptAttributeType)
 		}
 		return mappedClasses;
 	}
+
+	public Concept getConceptByRef(String conceptRef) throws APIException {
+		if(conceptRef == null) {
+			throw new APIException("null isn`t allowed as acceptable input");
+		}
+		conceptRef = conceptRef.trim();
+		if (conceptRef.contains(".")) {
+			return getConcept(evaluateStaticConstant(conceptRef));
+		}
+		Concept cpt = null;
+		int index = conceptRef.indexOf(":");
+		if (index != -1) {
+			String conceptSource, conceptCode;
+			conceptCode = conceptRef.substring(index+1);
+			conceptSource = conceptRef.substring(0, index);
+			cpt = Context.getConceptService().getConceptByMapping(conceptCode, conceptSource);
+			if (cpt != null)
+				return cpt;
+		}
+		if ( !(conceptRef.length() < 36 || conceptRef.length() > 38 || conceptRef.contains(" ") || conceptRef.contains("."))){
+			cpt = Context.getConceptService().getConceptByUuid(conceptRef);
+			if (cpt != null)
+				return cpt;
+		}
+		cpt = getConcept(conceptRef);
```

Comment:
What does this handle? `conceptId` or `name` or both?

And btw, you would just do:

```java
return getConcept(conceptRef);

```